moped-editor/src/utils/dateAndTime.js

```diff
@@ -114,3 +114,43 @@ export const formatRelativeDate = (timeStampTZString) => {
     return formatRelativeTime(millsElapsed, "day");
   }
 };
+
+/**
+ * A component that renders a formatted date string with primary display and optional secondary display
+ * @param {Object} props - The component props
+ * @param {string} props.date - The date to format, in a format parseable by new Date()
+ * @param {('relative'|'absolute'|'expanded'|'short'|'time'|'fullTime')} props.primary - The format to use for the primary display
+ * @param {('relative'|'absolute'|'expanded'|'short'|'time'|'fullTime')} [props.secondary] - The format to use for the secondary display, if any
+ * @returns {JSX.Element} A span element containing the formatted date string(s)
+ */
+export const FormattedDateString = ({ date, primary, secondary }) => {
+  const formatDate = (format) => {
```

Comment:
could you pull the definition of this function outside of this component so it isn't recreated on each render?